### PR TITLE
feat: add an option to change use a custom card overlay

### PR DESCRIPTION
**Pull request by osdnk**
I find it sometimes useful to define overlay renderer on my own. Eg. I needed to replace the background with BlurView and with this API I find it quite easy

### CODE REVIEW

File: packages/stack/src/types.tsx [hunk 1]
```diff
@@ -313,6 +313,10 @@ export type StackNavigationOptions = StackHeaderOptions &
       bottom?: number;
       left?: number;
     };
+    /**
+     * Function that given `style` returns a React Element to display as a overlay.
+     */
+    overlay?: (style: StyleProp<ViewStyle>) => React.ReactNode;
```
Comment (satya164) [inline packages/stack/src/types.tsx]: can we change it to an object with `style` property?
```suggestion
    overlay?: (style: StyleProp<ViewStyle>) => React.ReactNode;
```

Comment (osdnk) [reply packages/stack/src/types.tsx]: yes, it will be more intuitive

File: packages/stack/src/types.tsx [hunk 2]
```diff
@@ -313,6 +313,10 @@ export type StackNavigationOptions = StackHeaderOptions &
       bottom?: number;
       left?: number;
     };
+    /**
+     * Function that` returns a React Element to display as a overlay.
```
Comment (satya164) [inline packages/stack/src/types.tsx]:
```suggestion
     * Function that returns a React Element to display as a overlay.
```

File: packages/stack/src/views/Stack/Card.tsx [hunk 3]
```diff
@@ -471,12 +475,9 @@ export default class Card extends React.Component<Props> {
...
     return (
       <View pointerEvents="box-none" {...rest}>
-        {overlayEnabled && overlayStyle ? (
-          <Animated.View
-            pointerEvents="none"
-            style={[styles.overlay, overlayStyle]}
-          />
-        ) : null}
+        {overlayEnabled && overlayStyle
+          ? this.props.overlay(overlayStyle)
```
Comment (satya164) [inline packages/stack/src/views/Stack/Card.tsx]: The code doesn't match the type definitions (should be `({ style: overlayStyle })`)

**Pull request merged**